client/client.go

```diff
@@ -462,6 +462,46 @@ func (r *NotaryRepository) GetTargetByName(name string, roles ...string) (*Targe
 
 }
 
+// CheckTargetByName returns true if all passed-in roles have signed the specified target with the same hash.
+// If no roles are passed, it uses the targets role.  Note that child roles are considered if
+// we can't find the target in the specified role.
+func (r *NotaryRepository) CheckTargetByName(name string, roles ...string) (bool, error) {
+	if err := r.Update(false); err != nil {
+		return false, err
+	}
+
+	if len(roles) == 0 {
```

Comment:
Just checking the specified role seems clearer, but @endophage is correct that maybe you do care about the child roles.  

Since you can basically configure your delegation hierarchy however you want though, it's hard to guess what the exact API someone would need would be (maybe they care about checking only the specified roles). 

I'm fine with this function as is because it suits our purposes for now, but it does make some assumptions consistent with only one type of workflow, whereas a lot of the other code in client is more general.